src/ImpurityDensities/PtypePNJunctionImpurityDensity.jl

```diff
@@ -0,0 +1,59 @@
+"""
+    struct ThermalDiffusionLithiumDensity{T <: SSDFloat} <: AbstractImpurityDensity{T}
+
+A PN junctin impurity model based on lithium thermal diffusion and constant bulk impurity density.
+The surface lithium density is saturated.
+
+ref: [Dai _et al._ (2023)](https://doi.org/10.1016/j.apradiso.2022.110638)
+ 
+## Fields
+* `Li_annealing_temperature::T`: lithium annealing temperature when the lithium is diffused into the crystal. The default value is 623 K.
+* `Li_annealing_time::T`: lithium annealing time. The default value is 18 minutes.
+* `calculate_depth2surface::Function`: the function for describing the depth to surface.
+* `bulk_imp::T`: the bulk impurity density constant.
+* `bulk/surface_imp_model::T`: there two variables will be constructed with above parameters.
+"""
+
+struct PtypePNJunctionImpurityDensity{T <: SSDFloat} <: AbstractImpurityDensity{T}
+	Li_annealing_temperature::T
+	Li_annealing_time::T
+    calculate_depth2surface::Function
+    bulk_imp::T
+    surface_imp_model::ThermalDiffusionLithiumDensity{T}
+    bulk_imp_model::ConstantImpurityDensity{T}
+end
```

Comment:
As suggested before, I guess this could be generalized:
```suggestion
struct PtypePNJunctionImpurityDensity{T <: SSDFloat, ID <: AbstractImpurityDensity{T}} <: AbstractImpurityDensity{T}
	Li_annealing_temperature::T
	Li_annealing_time::T
    calculate_depth2surface::Function
    bulk_imp::T
    surface_imp_model::ThermalDiffusionLithiumDensity{T}
    bulk_imp_model::ID
end
```
as long as we allow for this (syntax-wise) in the config file.